user/serializers.py

```diff
@@ -1 +1,32 @@
 # write your code here
+from django.contrib.auth import get_user_model
+from rest_framework import serializers
+
+from user.models import User
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = get_user_model()
+        fields = ("id", "username", "email", "password", "is_staff")
+        read_only_fields = ("id", "is_staff")
+        extra_kwargs = {"password": {"write_only": True, "min_length": 5}}
+
+    def create(self, validated_data):
+        user = User(
+            email=validated_data["email"],
+            username=validated_data["username"]
+        )
+        user.set_password(validated_data["password"])
+        user.save()
+        return user
```

Comment:
you can use create_user method() to simplify the create method,  that handles the creation of user instances correctly, including password hashing.